source/testcommon/testcommon/dataframes/write_to_delta.py

```diff
@@ -0,0 +1,23 @@
+import os
+
+from pyspark.sql import types as T
+from pyspark.sql import SparkSession
+
+from testcommon.dataframes import read_csv
+
+def write_when_files_to_delta(
+    spark: SparkSession,
+    scenario_path: str,
+    files: list[tuple[str, T.StructType]]
+) -> None:
```

Comment:
Add docstring about how to use this. Prerequisites in the current implementation are tables must have been created in advance, and table names must have same name as the files. Except the extension. And how about multipart file names. Are they required or optional.

There is a lot of stuff that is not evident from the function signature alone.